engine.py

```diff
@@ -451,6 +454,106 @@ def _panels_supported(self):
+        commands_to_run = []
+
+        # Run the series of app instance commands listed in the 'run_at_startup'
+        # setting.
+        for app_setting_dict in self.get_setting("run_at_startup", []):
+
+            app_instance_name = app_setting_dict["app_instance"]
+
+            # Menu name of the command to run or '' to run all commands of the
+            # given app instance.
+            setting_cmd_name = app_setting_dict["name"]
+
+            # Retrieve the command dictionary of the given app instance.
+            cmd_dict = app_instance_commands.get(app_instance_name)
+
+            if cmd_dict is None:
+                self.log_warning(
+                    "%s configuration setting 'run_at_startup' requests app "
+                    "'%s' that is not installed." %
+                    (self.name, app_instance_name))
+            else:
+                if not setting_cmd_name:
+                    # add commands to the list for the given app instance.
+                    for (cmd_name, cmd_function) in cmd_dict.iteritems():
+                        self.log_debug(
+                            "%s startup running app '%s' command '%s'." %
+                            (self.name, app_instance_name, cmd_name)
+                        )
+                        commands_to_run.append(cmd_function)
+                else:
+                    # add commands whose name is listed in the 'run_at_startup'
+                    # setting.
+                    cmd_function = cmd_dict.get(setting_cmd_name)
+                    if cmd_function:
+                        self.log_debug(
+                            "%s startup running app '%s' command '%s'." %
+                            (self.name, app_instance_name, setting_cmd_name)
+                        )
+                        commands_to_run.append(cmd_function)
+                    else:
+                        known_commands = ', '.join(
+                            "'%s'" % name for name in cmd_dict)
+                        self.log_warning(
+                            "%s configuration setting 'run_at_startup' "
+                            "requests app '%s' unknown command '%s'. Known "
+                            "commands: %s" % (
+                                self.name,
+                                app_instance_name,
+                                setting_cmd_name,
+                                known_commands
+                            )
+                        )
+
+        # no commands to run. just bail
+        if not commands_to_run:
+            return
+
+        # here we wrap the commands to run in a single function we can hand over
+        # to houdini as an event loop callback. This will run when houdini is
+        # idle which should be after the UI loads up.
+        def run_when_idle():
+            for command in commands_to_run:
+                # iterate over all the commands and execute them.
+                command()
```

Comment:
perhaps a debug message here? Or is that maybe done already as part of the callback?